kbase-extension/static/kbase/js/widgets/narrative_core/catalog/app_card.js

```diff
@@ -1,341 +1,331 @@
                 for(var k=0; k<this.$divs.length; k++) {
-                    this.$divs[k].find('.kbcb-runs').empty()
-                        .append('<i class="fa fa-share"></i>')
-                        .append($('<span>').addClass('kbcb-run-count').append(this.runCount))
-                        .tooltip({title:'Ran in a Narrative '+this.runCount+' times.', placement:'bottom',container:'body',
-                                        delay:{show: 400, hide: 40}});;
+                    this.$divs[k].find('.kbcb-star')
+                        .removeClass('kbcb-star-favorite').removeClass('kbcb-star-nonfavorite');
                 }
-            }
-        };
-        this.getRunCount = function() {
-            if(this.runCount) return this.runCount;
-            return 0;
-        }
-
+            };
+
+            this.getStarCount = function() {
+                if(this.starCount) return this.starCount;
+                return 0;
+            };
+
+            this.setStarCount = function(count) {
+                this.starCount = count;
+                if(this.starCount<=0) { this.starCount = null; }
+                if(this.starCount) {
+                    for(let k=0; k<this.$divs.length; k++) {
+                        this.$divs[k].find('.kbcb-star-count').html(count);
+                    }
+                } else {
+                    for(let k=0; k<this.$divs.length; k++) {
+                        this.$divs[k].find('.kbcb-star-count').empty();
+                    }
+                }
+            };
+
+            this.runCount = null;
+
+            this.setRunCount = function(runs) {
+                this.runCount = runs;
+                if(this.runCount) {
+                    for(var k=0; k<this.$divs.length; k++) {
+                        this.$divs[k].find('.kbcb-runs').empty()
+                            .append('<i class="fa fa-share"></i>')
+                            .append($('<span>').addClass('kbcb-run-count').append(this.runCount))
+                            .tooltip({title:'Ran in a Narrative '+this.runCount+' times.', placement:'bottom',container:'body',
+                                delay:{show: 400, hide: 40}});
+                    }
+                }
+            };
+            this.getRunCount = function() {
+                if(this.runCount) return this.runCount;
+                return 0;
+            };
 
-        /* rendering methods that are shared in multiple places */
-        this._renderAppCard = function() {
 
-            var info = this.info;
-            var type = this.type;
-            var tag = this.tag;
-            var nms_base_url = this.nms_base_url;
+            /* rendering methods that are shared in multiple places */
+            this._renderAppCard = function() {
 
-            // Main Container
-            var $appDiv = $('<div>').addClass('kbcb-app-card kbcb-hover container');
+                var info = this.info;
+                var self = this;
```

Comment:
Removed variables here (`type`, `tag`, `mms_base_url`) that were shadowing declarations in the upper level, and hoisted the `var self` declaration up to the start of the function (which is what JS would do).